packages/components/src/date-time/test/time.js

```diff
@@ -47,13 +47,13 @@ describe( 'TimePicker', () => {
 		fireEvent.change( hoursInput, { target: { value: '12' } } );
 		fireEvent.blur( hoursInput );
 
-		expect( onChangeSpy ).toHaveBeenCalledWith( '2018-12-22T12:00:00' );
+		expect( onChangeSpy ).toHaveBeenCalledWith( '2018-12-22T00:00:00' );
 		onChangeSpy.mockClear();
 
 		fireEvent.change( minutesInput, { target: { value: '35' } } );
 		fireEvent.blur( minutesInput );
 
-		expect( onChangeSpy ).toHaveBeenCalledWith( '2018-12-22T12:35:00' );
+		expect( onChangeSpy ).toHaveBeenCalledWith( '2018-12-22T00:35:00' );
```

Comment:
The fix introduced in this PR caused this unit test to fail. After looking into it more, I believe that the unit test was previously wrong:

- the time picker in the test is using the 12 hours period
- the initial time set to the picker is `1986-10-18T11:00:00`, and therefore the picker automatically picks `AM` as the period
- the period is not changed explicitly throughout the test
- the `hours` input's value is changed to `12` (remember, the period is still set to `am`)
- the test currently expected `12am` to be converted to midday
- the test was passing until now because of the lack of value adjustment prior to this PR

I believe that it is correct to update the unit test in this case, since `12am` should be converted to `00` in the 24h format

Reply:
I think so too.

Reply:
Yes, +1 from me!